pkg/koordlet/resmanager/cpu_suppress.go

```diff
@@ -341,20 +332,68 @@ func (r *CPUSuppress) suppressBECPU() {
 		r.suppressPolicyStatuses[string(slov1alpha1.CPUCfsQuotaPolicy)] = policyUsing
 		r.recoverCPUSetIfNeed()
 	} else {
-		adjustByCPUSet(suppressCPUQuantity, nodeCPUInfo)
+		r.adjustByCPUSet(suppressCPUQuantity, nodeCPUInfo)
 		r.suppressPolicyStatuses[string(slov1alpha1.CPUSetPolicy)] = policyUsing
 		r.recoverCFSQuotaIfNeed()
 	}
 }
 
-func adjustByCPUSet(cpusetQuantity *resource.Quantity, nodeCPUInfo *metriccache.NodeCPUInfo) {
+func (r *CPUSuppress) adjustByCPUSet(cpusetQuantity *resource.Quantity, nodeCPUInfo *metriccache.NodeCPUInfo) {
 	oldCPUSet, err := util.GetRootCgroupCurCPUSet(corev1.PodQOSBestEffort)
 	if err != nil {
 		klog.Warningf("applyBESuppressPolicy failed to get current best-effort cgroup cpuset, err: %s", err)
 		return
 	}
 
-	beCPUSet := calculateBESuppressCPUSetPolicy(cpusetQuantity, len(oldCPUSet), nodeCPUInfo)
+	podMetas := r.resmanager.statesInformer.GetAllPods()
+	// value: 0 -> lse, 1 -> lsr, not exists -> others
+	cpuIdToPool := map[int32]apiext.QoSClass{}
+	for _, podMeta := range podMetas {
+		alloc, err := apiext.GetResourceStatus(podMeta.Pod.Annotations)
+		if err != nil {
+			continue
+		}
+		if alloc.CPUSet != "" {
+			set, err := cpuset.Parse(alloc.CPUSet)
+			if err != nil {
+				klog.Errorf("failed to parse cpuset info of pod %s, err: %v", podMeta.Pod.Name, err)
+				continue
+			}
+			for _, cpuID := range set.ToSliceNoSort() {
+				cpuIdToPool[int32(cpuID)] = apiext.GetPodQoSClass(podMeta.Pod)
+			}
+		}
+	}
+	lsrCpus := []util.ProcessorInfo{}
+	lsCpus := []util.ProcessorInfo{}
+	for _, processor := range nodeCPUInfo.ProcessorInfos {
```

Comment:
please add a FIXME comments here: "be pods might be starved since lse pods can run out of all cpus"